templates/templates_test.go

```diff
@@ -0,0 +1,71 @@
+package templates
+
+import (
+	"encoding/json"
+	"github.com/sensu/sensu-go/types"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var (
+	templateOk          = "Check: {{ .Check.Name }} Entity: {{ .Entity.Name }} !"
+	templateVarNotFound = "Check: {{ .Check.NameZZZ }} Entity: {{ .Entity.Name }} !"
+	templateInvalid     = "Check: {{ .Check.Name Entity: {{ .Entity.Name }} !"
+)
+
+// Valid test
+func TestEvalTemplate_Valid(t *testing.T) {
+	reader, _ := os.Open("test/event-check-entity-override.json")
```

Comment:
Reading an external file could cause some test stability issues. Is the file mandatory for this use case? Or perhaps could you define `event-check-entity-override.json` as a byte array:
```
[]byte(`{"timestamp": 1550816106, "entity": {"entity_class": "agent", "system": ...`)
```

Reply:
Thanks for pointing that out, I wasn't aware there could be problems using files when running unit tests. I was able to run the test successfully both in my IDE and from the command line. However I'm not too familiar on how the build pipeline will handle this.

The event JSON content is required for the test so I took your suggestion and moved it to a Go file as a `[]byte`. It'll now get compiled in the test itself.